app/models/trip.rb

```diff
@@ -0,0 +1,18 @@
+class Trip < ApplicationRecord
+  before_save :calculate_distance
```

Comment:
I'm not sure if doing this `before_save` is the best solution. I would prefer to do this as a normal method call rather than as a callback.